tools/cat2table.ml

```diff
@@ -0,0 +1,1107 @@
+    | _ -> (defs, parsed_files)
+
+  and tr_bds defs kont bds =
+    List.fold_left
+      (fun defs (_, p, expr) ->
+        match p with Pvar (Some id) -> kont defs id expr | _ -> defs)
+      defs bds
+
+  let tr_ast fname e1 e2 start_rel =
+    let num_defs, _ =
+      tr_ast StringMap.empty
+        (fun _ num_defs id _ ->
+          let num = StringMap.safe_find 0 id num_defs in
+          StringMap.add id (num + 1) num_defs)
+        StringSet.empty fname
+    in
+    let (defs, _), _ =
+      tr_ast
+        (StringMap.empty, StringMap.empty)
+        (fun is_rec (defs, crt_nums) id expr ->
+          let converted_def = convert_defs num_defs crt_nums id is_rec expr in
+          let num = StringMap.safe_find 0 id crt_nums + 1 in
+          let total_num = StringMap.find id num_defs in
+          let crt_nums = StringMap.add id num crt_nums in
+          let id = if num = total_num then id else to_alias id num in
+          let defs = StringMap.add id converted_def defs in
+          (defs, crt_nums))
+        StringSet.empty fname
+    in
+    let defs =
+      StringMap.map
+        (fun def -> replace_ifs_and_tries (ASTUtils.flatten def))
+        defs
+    in
+    match (e1, e2) with
+    | Some e1, Some e2 ->
+        let e1 = parse_effect_from_string "e1" e1 |> ASTUtils.flatten in
+        let e2 = parse_effect_from_string "e2" e2 |> ASTUtils.flatten in
+        let sol = tr_rel e1 e2 start_rel defs in
+        if sol = ExprGraph.empty then Printf.printf "No solutions\n"
+        else
+          ignore
+            (ExprGraph.dfs_fold
+               ~per_vertex:(fun rel lvl _ ->
+                 Printf.printf "%s%s\n" (String.make lvl '-') (pp_expr rel))
+               ~per_neighbour:(fun _ _ _ -> ())
+               sol (make_var start_rel) ExprGraph.VSet.empty ())
+    | _ -> Warn.fatal "Omission of e1 or e2 not implemented yet"
+end
+
+let verbose = ref false
+let libdir = ref (Filename.concat Version.libdir "herd")
+let includes = ref []
+let model = ref "herd/libdir/aarch64.cat"
+let axioms_file = ref "herd/libdir/aarch64axioms.cat"
+let e1 = ref None
+let e2 = ref None
+let start_rel = ref "ob"
+
+let options =
+  [
+    ( "-version",
+      Arg.Unit
+        (fun () ->
+          Printf.printf "%s, Rev: %s\n" Version.version Version.rev;
+          exit 0),
+      " show version number and exit" );
+    ( "-libdir",
+      Arg.Unit
+        (fun () ->
+          print_endline !libdir;
+          exit 0),
+      " show installation directory and exit" );
+    ( "-set-libdir",
+      Arg.String (fun s -> libdir := s),
+      "<path> set installation directory to <path>" );
+    ArgUtils.parse_string "-model" model !model;
+    ArgUtils.parse_string "-axioms_file" axioms_file !axioms_file;
+    ("-e1", Arg.String (fun s -> e1 := Some s), "<str> first effect");
+    ("-e2", Arg.String (fun s -> e2 := Some s), "<str> second effect");
+    ("-start", Arg.String (fun s -> start_rel := s), "<str> top level relation");
```

Comment:
The help message here should indicate that `ob` (or, whatever `start_rel` holds by default) is the default value.